dali/operators/reader/loader/video/frames_decoder_gpu.cc

```diff
@@ -40,28 +40,62 @@ int process_picture_decode(void *user_data, CUVIDPICPARAMS *picture_params) {
 }
 }  // namespace detail
 
+
+void FramesDecoderGpu::InitBitStreamFilter() {
+  const AVBitStreamFilter *bsf = nullptr;
+
+  switch (av_state_->codec_->id) {
+  case AVCodecID::AV_CODEC_ID_H264:
+    bsf = av_bsf_get_by_name("h264_mp4toannexb");
+    break;
+  case AVCodecID::AV_CODEC_ID_HEVC:
+    bsf = av_bsf_get_by_name("hevc_mp4toannexb");
+    break;
+  default:
+    DALI_FAIL(make_string(
+      "Could not find suitable bit stream filter for codec: ",
+      av_state_->codec_->name));
+  }
+
+  DALI_ENFORCE(
+    av_bsf_alloc(bsf, &bsfc_) >= 0,
+    "Unable to allocate bit stream filter");
+  DALI_ENFORCE(
+    avcodec_parameters_copy(bsfc_->par_in, av_state_->ctx_->streams[0]->codecpar) >= 0,
+    "Unable to copy bit stream filter parameters");
+  DALI_ENFORCE(
+    av_bsf_init(bsfc_) >= 0,
+    "Unable to initialize bit stream filter");
+}
+
+cudaVideoCodec FramesDecoderGpu::GetCodecType() {
+  if (av_state_->codec_->id == AV_CODEC_ID_HEVC) {
```

Comment:
Just an idea, if it is validated beforehand you could add the `assert(av_state_->codec_->id == AV_CODEC_ID_HEVC || av_state_->codec_->id == AV_CODEC_ID_H264)` here for future self to highlight the assumption.

Reply:
Added comment: `// Code assumes av_state_->codec_->id in FramesDecoder::SupportedCodecs` there